Resources/assets/js/adapter/fancytree.js

```diff
@@ -0,0 +1,234 @@
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import Map from 'core-js/es6/map'
+import '../jquery.cmf_context_menu'
+import 'fancytree/jquery.fancytree.js'
+import 'fancytree/skin-win8/ui.fancytree.css'
+import '../../css/fontawesome-style.css'
+
+var cache = new Map();
+
+function getPropertyFromString(propertyPath, list) {
+    var isOptional = propertyPath.substr(0, 1) === '?';
+    var props = propertyPath.substr(1).split('.');
+    var currentNode = list;
+    for (prop in props) {
+        currentNode = currentNode[props[prop]];
+
+        if (undefined === currentNode) {
+            if (isOptional) {
+                break;
+            }
+
+            throw 'Attribute "' + props[prop] + '" does not exists';
+        }
+    }
+
+    return currentNode;
+}
+
+/**
+ * A tree browser adapter for the Fancytree library.
+ *
+ * @author Wouter J <wouter@wouterj.nl>
+ * @see https://github.com/mar10/fancytree
+ */
+export class FancytreeAdapter {
+    constructor(options) {
+        if (!window.jQuery || !jQuery.fn.fancytree) {
+            throw 'The FancytreeAdapter requires both jQuery and the FancyTree library.';
+        }
+
+        if (!options.request) {
+            throw 'The FancytreeAdapter requires a request option.';
+        }
+
+        this.requestData = options.request;
+        this.rootNode = options.root_node || '/';
+        this.useCache = undefined === options.use_cache ? true : options.use_cache;
+
+        /**
+         * The available actions.
+         *
+         * @var array
+         */
+        this.actions = new Map();
+        /**
```

Comment:
If we assume PHP coding standards than we are missing an empty line before these doc blocks.